### ISSUE TITLE
Introduce analyzer or some other mechanism to flag builds that invoke the compiler (i.e. CoreCompile target) with same inputs and outputs to avoid race conditions

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/msbuild#9781]
Similar to https://github.com/dotnet/msbuild/issues/9553

https://github.com/dotnet/command-line-api/pull/2336 is a good example to talk about the issue. `System.CommandLine.csproj` was built twice because two additional global properties were passed in when referencing it from a specific test project. The global properties didn't affect the CoreCompile target and the compiled output at all. This resulted in a race condition as two processes were trying to copy to the same output at the same time.

99% of the time this is unintentional and it would be great if msbuild could flag that somehow via an analyzer or some other means.

**Comment by rainersigwald** [comment]
Yeah this is my top analyzer request. @JanKrivanek do we not have a tracking bug? I would have sworn we did but don't see one.

**Comment by JanKrivanek** [comment]
Adding to https://github.com/dotnet/msbuild/issues/9630
